### PR TITLE
API Level auth providers as an Array

### LINKED ISSUE #915: API Level auth providers not working properly
### Description:
Security scheme defined in API level not working.

### Steps to reproduce:
Add similar to the following to a valid API definition and build a microgateway project using that definition.
`security:
  - mybasic: []
components:
  securitySchemes:
    mybasic:
      type: http
      scheme: basic
`
Basic auth is not enabled at API level but only default security "jwt","oauth2" can be observed.

### Affected Product Version:
jBallerina

### Environment details (with versions):
- OS: macos catalina
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Purpose
Provide API Level auth providers as an Array to the services.mustache

### Issues

Fixes #915

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

catalina 
java version "1.8.0_201"

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java [hunk 1]
```diff
@@ -951,4 +952,19 @@ private static boolean isResourceEpUnavailable(Operation operation) {
...
+
+    public static List<String> setAuthProviders(BasicAuth basicAuth) {
+        List<String> authProviders = new ArrayList<>();
+        if (basicAuth != null && basicAuth.isOptional()) {
+            authProviders.add(APISecurity.basic.name());
+            authProviders.add(APISecurity.oauth2.name());
+            authProviders.add(APISecurity.jwt.name());
+        } else if (basicAuth != null && basicAuth.isRequired()) {
+            authProviders.add(APISecurity.basic.name());
+        } else {
```
Comment (Rajith90) [inline components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java]: Lets remove this logic. We need to honour the values comes from the swagger.  If the security definitions are not defined by default oauth and jwt should be added.

**Comment by praminda** [comment]
Oops merged the PR. Lets add the commented improvement to a new PR

File: components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java (hunk 1 above)
Comment (AmaliMatharaarachchi) [reply components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/utils/OpenAPICodegenUtils.java]: Fixed in #932

**Comment by AmaliMatharaarachchi** [comment]
@praminda It is fixed in #932

**Pull request merged**